backend/lcfs/web/api/transaction/repo.py

```diff
@@ -67,33 +67,26 @@ async def get_transactions_paginated(self, offset: int, limit: int, conditions:
                     [status.value for status in visible_statuses_transferee])
             )
 
-            # TODO: Additional visibility checks needed for other transaction types.
-            # For transactions that are not of type "Transfer", include them without visibility filtering
+            # Conditions for non-transfer transactions, only "Approved" for suppliers
             non_transfer_condition = and_(
                 TransactionView.transaction_type != 'Transfer',
-                TransactionView.to_organization_id == organization_id
+                TransactionView.to_organization_id == organization_id,
+                TransactionView.status == 'Approved' # This status includes InitiativeAgreement and AdminAdjustment
             )
 
             # Combine conditions since an organization can be both transferor and transferee, or neither for non-"Transfer" transactions
             combined_role_condition = or_(
                 transferor_condition, transferee_condition, non_transfer_condition)
             query_conditions.append(combined_role_condition)
         else:
-            # Fetch visible statuses for government, but only for "Transfer" transactions
-            visible_statuses_government = await self.get_visible_statuses(EntityType.Government)
-            government_condition = and_(
+            # Government view should see all non-transfer transactions regardless of status
+            gov_transfer_condition = and_(
                 transfer_type_condition,
-                TransactionView.status.in_(
-                    [status.value for status in visible_statuses_government])
+                TransactionView.status.in_([status.value for status in await self.get_visible_statuses(EntityType.Government)])
```

Comment:
Combining statuses is causing duplicates.
Transaction statuses in the filter drop down of ag-grid is showing duplicate values.